jwst/extract_2d/grisms.py

```diff
@@ -89,10 +89,15 @@ def extract_tso_object(input_model,
     if extract_orders is None:
         log.info("Using default order extraction from reference file")
         extract_orders = ref_extract_orders
+        available_orders = [x[1] for x in extract_orders if x[0] == input_model.meta.instrument.filter].pop()
     else:
-        raise NotImplementedError("Multiple order extraction for TSO not currently implemented")
+        if not isinstance(extract_orders, list):
+            raise TypeError('Expected extract_orders to be a list of integers')
+        print(extract_orders)
+        available_orders = extract_orders
 
-    available_orders = [x[1] for x in extract_orders if x[0] == input_model.meta.instrument.filter].pop()
+    if len(available_orders) > 1:
+        raise NotImplementedError("Multiple order extraction for TSO not currently implemented")
```

Comment:
Just curious as to why we can't do multiple orders for TSO mode? Is it a problem with not knowing where the other orders are located or what?

Reply:
The team explicitly does not want anything but the first order subtracted, so the output from the step is a `SlitModel` only, not a `MultiSlitModel`... This is reinforced in the reference file. However, I left the information there for the second order, and wanted to make it possible for someone to choose to try and extract the second order if they wanted, it just needs to create two different output files.  I can add in creation of a `MultiSlitModel` if in the future the team wants to extract multiple orders at the same time, but that's not something they've asked for and I'm working hard at delivering requirements w/o too many unasked for bells and whistles ;)